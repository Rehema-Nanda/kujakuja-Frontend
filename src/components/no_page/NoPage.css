.no-page {
    text-align: center;
    min-height: calc(100vh - 248px);
}

.no-page-container {
    width: 100%;
    top: 145px;
    background: #FFFFFF;
    box-shadow: 0px 8px 12px rgba(23, 29, 33, 0.14);
    border-radius: 12px;
    margin-top: 145px;
}

.no-page-content {
    display: block;
    padding-top: 50px;
    padding-right: 46px;
    padding-left: 48px;
    font-family: Open Sans;
    font-style: normal;
    font-weight: normal;
    font-size: 16px;
    line-height: 24px;
}

.no-page-buttons {
    width: 100%;
    height: 56px;
    top: 251px;
    background: #F5F5F5;
    border-radius: 0px 0px 12px 12px;
}

.no-page-option {
    padding-top: 16px;
}

.no-page-back {
    cursor: pointer;
}

a, a:hover {
    color: black;
    text-decoration: none;
}

.no-page-icon {
    width: 73px;
    height: 73px;
    position: absolute;
    top: 106px;
    left: calc(50% - 36px);
    border-radius: 50%;
}

.no-page-icon img {
    height: 92px;
    position: relative;
    top: -8px;
}
