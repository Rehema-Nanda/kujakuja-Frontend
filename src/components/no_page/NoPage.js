
import React from "react";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { faArrowLeft, faArrowRight } from "@fortawesome/free-solid-svg-icons";
import { Row, Col, Container } from "reactstrap";
import i18n from "i18next";

import GlobalFooter from "../GlobalFooter";
import "./NoPage.css";

const facePath = require("../../img/face_unsatisfied.png");

const NoPage = ({ history }) => {
    return (
        <div>
            <div className="no-page">
                <div className="title-background-parent">
                    <div className="title-background-shape" />
                </div>
                <Container>
                    <Row>
                        <Col sm="12" md={{ size: 6, offset: 3 }}>
                            <div className="no-page-container">
                                <div className="no-page-icon"><img src={facePath} alt="alt" /></div>
                                <div className="no-page-content">
                                    <h2>{i18n.t("noPage.title")}</h2>
                                    <p>{i18n.t("noPage.description")}</p>
                                </div>
                                <div className="no-page-buttons">
                                    <Row>
                                        <Col xs="6" className="no-page-option">
                                            <span className="no-page-back" onClick={history.goBack}>
                                                <FontAwesomeIcon icon={faArrowLeft} />
                                                {" "}
                                                {i18n.t("noPage.goback")}
                                            </span>
                                        </Col>
                                        <Col xs="6" className="no-page-option">
                                            <Link to="/datamap">
                                                <span className="no-page-back">
                                                    {i18n.t("noPage.home")}
                                                    {" "}
                                                    <FontAwesomeIcon icon={faArrowRight} />
                                                </span>
                                            </Link>
                                        </Col>
                                    </Row>
                                </div>
                            </div>
                        </Col>
                    </Row>
                </Container>
            </div>
            <GlobalFooter />
        </div>
    );
};

export default NoPage;
