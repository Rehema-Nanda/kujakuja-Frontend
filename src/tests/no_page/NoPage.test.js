import React from "react";
import { mount } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import NoPage from "../../components/no_page/NoPage";

describe("NoPage component", () => {
    const testProps = {
        history: {
            goBack: jest.fn(),
        },
    };
    const wrapper = mount(<BrowserRouter><NoPage {...testProps} /></BrowserRouter>);

    it("render correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
