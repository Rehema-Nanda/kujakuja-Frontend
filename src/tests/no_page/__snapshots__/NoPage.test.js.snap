// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoPage component render correctly 1`] = `
"<BrowserRouter>
  <Router history={{...}}>
    <NoPage history={{...}}>
      <div>
        <div className=\\"no-page\\">
          <div className=\\"title-background-parent\\">
            <div className=\\"title-background-shape\\" />
          </div>
          <Container tag=\\"div\\">
            <div className=\\"container\\">
              <Row tag=\\"div\\">
                <div className=\\"row\\">
                  <Col sm=\\"12\\" md={{...}} tag=\\"div\\" widths={{...}}>
                    <div className=\\"col-sm-12 col-md-6 offset-md-3\\">
                      <div className=\\"no-page-container\\">
                        <div className=\\"no-page-icon\\">
                          <img src=\\"face_unsatisfied.png\\" alt=\\"alt\\" />
                        </div>
                        <div className=\\"no-page-content\\">
                          <h2 />
                          <p />
                        </div>
                        <div className=\\"no-page-buttons\\">
                          <Row tag=\\"div\\">
                            <div className=\\"row\\">
                              <Col xs=\\"6\\" className=\\"no-page-option\\" tag=\\"div\\" widths={{...}}>
                                <div className=\\"no-page-option col-6\\">
                                  <span className=\\"no-page-back\\" onClick={[Function: mockConstructor]}>
                                    <FontAwesomeIcon icon={{...}}>
                                      <i className=\\"fa\\" />
                                    </FontAwesomeIcon>
                                     
                                  </span>
                                </div>
                              </Col>
                              <Col xs=\\"6\\" className=\\"no-page-option\\" tag=\\"div\\" widths={{...}}>
                                <div className=\\"no-page-option col-6\\">
                                  <Link to=\\"/datamap\\">
                                    <LinkAnchor href=\\"/datamap\\" navigate={[Function: navigate]}>
                                      <a href=\\"/datamap\\" onClick={[Function: onClick]}>
                                        <span className=\\"no-page-back\\">
                                           
                                          <FontAwesomeIcon icon={{...}}>
                                            <i className=\\"fa\\" />
                                          </FontAwesomeIcon>
                                        </span>
                                      </a>
                                    </LinkAnchor>
                                  </Link>
                                </div>
                              </Col>
                            </div>
                          </Row>
                        </div>
                      </div>
                    </div>
                  </Col>
                </div>
              </Row>
            </div>
          </Container>
        </div>
        <GlobalFooter>
          <section className=\\"footer\\">
            <div className=\\"footer-details\\">
              <div className=\\"footer-logo\\">
                <img src={{...}} alt=\\"idea feed\\" />
              </div>
              <div className=\\"footer-social\\">
                <div className=\\"social-instagram\\" onClick={[Function: onClick]} />
                <div className=\\"social-fb\\" onClick={[Function: onClick]} />
                <div className=\\"social-twitter\\" onClick={[Function: onClick]} />
                <div className=\\"social-medium\\" onClick={[Function: onClick]} />
              </div>
            </div>
          </section>
        </GlobalFooter>
      </div>
    </NoPage>
  </Router>
</BrowserRouter>"
`;
